### PR TITLE
fix: routine parameters mismatch bug

### LINKED ISSUE #74: Bug: Receiving an error stating "ERROR_INVALID_ARGUMENT" even after providing all necessary valid arguments.
### Description
- When attempting to invoke an endpoint through the js-moi-sdk, an error stating "One or more required arguments are missing" is encountered despite all required arguments being provided.

### Contract
```
endpoint invoke persistent addProof(
        moiID Address, 
        IDtype []String, 
        country String, 
        status String, 
        time U256, 
        cid String, 
        verifier []Bytes
    ):
        mutate verificationProofs <- verfieye.verificationProofs:
                if verificationProofs[moiID]?:
                    throw "Given moiID already has a proof exist"
                
                memory proof = (proofDetails) <- createProof(moiID: moiID, IDtype: IDtype, country: country, status: status, time: time, cid: cid, verifier: verifier)
                memory local_proof Map[Address]Proof
                local_proof[moiID] = proof
                disperse verificationProofs <- local_proof
```

### TS code
```
private async initWallet(): Promise<moi.Wallet> {
    const wallet = await moi.Wallet.fromMnemonic(mnemonic, keyPath);
    wallet.connect(provider);
    return wallet;
}

private async getLogicDriver(): Promise<moi.LogicDriver<any>> {
    const wallet = await this.initWallet();
    return await moi.getLogicDriver(logicID, wallet);
}
async recordVerification(moiID: string, details: VerificationDetails): Promise<void> {
        try {
            const logic = await this.getLogicDriver();
            const { docTypes: IDtype, country, status, proofDetails } = details;
            const { id: cid, meta: verifier } = proofDetails;
            const time = Date.now();

            const ix = await logic.routines.addProof(moiID, IDtype, country, status, time, cid, verifier);
            console.log(ix.hash);
            const receipt = await ix.wait();
            console.log(receipt);
        } catch (err) {
            console.error(err);
            throw new Error("Failed to record verification");
        }
    }
```

### Manifest
[verifeye.json](https://github.com/user-attachments/files/16278319/verifeye.json)

---
**Expected Output**: Should get the result and hash.

**Actual Output**: 
```
 code: 'ERROR_INVALID_ARGUMENT',
  reason: 'One or more required arguments are missing.',
  params: {}
```

### PR DESCRIPTION
**Pull request by 0xharshrastogi**
# Fix bug "ERROR_INVALID_ARGUMENT" even after providing all necessary arguments

## Description

This pull request addresses the issue described in #74. The bug occurred due to the last parameter for the routine being a `Uint8Array` of type `object`. Same can happen with other types values `Map` or `class`. To resolve this, a new class `RoutineOptions` was created to validate the options. Additionally, a new `createRoutineOption` function has been exposed to facilitate the creation of instances of `RoutineOptions`.

- Created new class `RoutineOptions` to validate routine options.
- Exposed `createRoutineOption` function to easily create instances of `RoutineOptions`.

- Closes #74 #84

## Changes Include

- [X] Bugfix (non-breaking change that solves an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

## Checklist

- [X] I have assigned this PR to myself
- [X] I have added at least 1 reviewer
- [X] I have tested this code
- [ ] I have updated the README and other relevant documents (guides...)

### CODE REVIEW

**Comment by sarvalabs-pawankalyan** [comment]
Works as expected

File: packages/js-moi-logic/src.ts/logic-base.ts [hunk 1]
```diff
@@ -217,9 +214,7 @@ export abstract class LogicBase extends ElementDescriptor {
         }
 
         ixObject.send = async (): Promise<InteractionResponse> => {
-            option.fuelLimit = option.fuelLimit != null ? option.fuelLimit : await ixObject.estimateFuel();
-            option.fuelPrice = option.fuelPrice != null ? option.fuelPrice : DEFAULT_FUEL_PRICE;
```
Comment (sarvalabs-gokul) [inline packages/js-moi-logic/src.ts/logic-base.ts:221]: why did we remove fuelPrice logic are we handling it in some other place.

File: packages/js-moi-logic/src.ts/logic-driver.ts [hunk 2]
```diff
@@ -59,9 +60,9 @@ export class LogicDriver<T extends Record<string, (...args: any) => any> = any>
                 return;
             }
 
-            routines[routine.name] = async (...params: any[]) => {
+            routines[routine.name] = async (...params: [...args: any[], options: RoutineOption]) => {
```
Comment (sarvalabs-gokul) [inline packages/js-moi-logic/src.ts/logic-driver.ts:63]: isn't options optional?

**Review by sarvalabs-gokul — CHANGES_REQUESTED** [review]

File: packages/js-moi-logic/src.ts/logic-base.ts [hunk 3]
```diff
@@ -185,6 +173,16 @@ export abstract class LogicBase extends ElementDescriptor {
     protected createIxRequest(ixObject: LogicIxObject): LogicIxRequest {
         const unwrap = async () => {
             const ix = await ixObject.call();
+            const error =
+              "error" in ix.receipt.extra_data &&
+              ix.receipt.extra_data.error != "0x"
+                ? ManifestCoder.decodeException(ix.receipt.extra_data.error)
```
Comment (sarvalabs-gokul) [inline packages/js-moi-logic/src.ts/logic-base.ts:179]: decode exception itself check for` ix.receipt.extra_data.error != "0x"`.

**Review by sarvalabs-gokul — CHANGES_REQUESTED** [review]

File: packages/js-moi-logic/src.ts/logic-base.ts (hunk 1 above)
Comment (sarvalabs-harshrastogi) [reply packages/js-moi-logic/src.ts/logic-base.ts:221]: Resolved In : https://github.com/sarvalabs/js-moi-sdk/pull/75/commits/13b0708ef769be3ddb23de85975b805f491f7163

File: packages/js-moi-logic/src.ts/logic-driver.ts (hunk 2 above)
Comment (sarvalabs-harshrastogi) [reply packages/js-moi-logic/src.ts/logic-driver.ts]: https://github.com/sarvalabs/js-moi-sdk/pull/75/commits/13b0708ef769be3ddb23de85975b805f491f7163

File: packages/js-moi-logic/src.ts/logic-base.ts (hunk 3 above)
Comment (sarvalabs-harshrastogi) [reply packages/js-moi-logic/src.ts/logic-base.ts]: https://github.com/sarvalabs/js-moi-sdk/pull/75/commits/13b0708ef769be3ddb23de85975b805f491f7163

**Review by sarvalabs-gokul — APPROVED** [review]

**Pull request merged**